src/kernel/chain.cpp

```diff
@@ -25,3 +26,13 @@ interfaces::BlockInfo MakeBlockInfo(const CBlockIndex* index, const CBlock* data
     return info;
 }
 } // namespace kernel
+
+std::ostream& operator<<(std::ostream& os, const ChainstateRole& role) {
+    switch(role) {
+        case ChainstateRole::NORMAL: os << "normal"; break;
+        case ChainstateRole::ASSUMEDVALID: os << "assumedvalid"; break;
+        case ChainstateRole::BACKGROUND: os << "background"; break;
+        default: os.setstate(std::ios_base::failbit);
```

Comment:
This would be resolved by @ryanofsky 's effort to remove the chainstate names/types entirely, as discussed here and linked at the bottom: https://github.com/bitcoin/bitcoin/pull/28562#discussion_r1344824078